### DISCUSSION TITLE
How can calculate a network size for Ai deck?

### DISCUSSION THREAD

**Discussion by godhj93** [bitcraze/discussions#585] [Q&A]
Hi, I am quite new embedded systems which have limited memory.

I want to calculate the number of parameters of my custom network for deploying it to ai deck.

but it seems that the ai deck's memory is depends on the input size as well.

Can I get some guidance for that?

Thanks :)

**Comment by hmllr** [comment]
Hi :)
Unfortunately I am not sure what you are asking - the number of parameters in a CNN is always dependent on the input size?
There are lots of examples if you google for how to compute the number of parameters (as for example https://towardsdatascience.com/understanding-and-calculating-the-number-of-parameters-in-convolution-neural-networks-cnns-fc88790d530d). 
For the space those parameters then need on the AIdeck's memory it is important in which format they are saved - usually you'd want to quantize it to 8bits, so you'll need 1 byte per parameter. In L2 you of course also need space for the current and next activations.
Hope this helps,
Hanna
